x-pack/plugins/data_enhanced/public/search/async_search_strategy.ts

```diff
@@ -51,27 +51,34 @@ export function asyncSearchStrategyProvider(
         )
       : NEVER;
 
-    return (syncSearch.search(request, options) as Observable<IAsyncSearchResponse>).pipe(
-      expand((response) => {
-        // If the response indicates of an error, stop polling and complete the observable
-        if (!response || (response.is_partial && !response.is_running)) {
-          return throwError(new AbortError());
-        }
+    return startServices$.pipe(
+      flatMap((startServices) => {
+        const syncSearch = (startServices[1] as DataEnhancedStartDependencies).data.search.getSearchStrategy(
```

Comment:
You shouldn't have to cast for any of these, I believe you can just do something like `CoreSetup<DataEnhancedStartDependencies>` for what you pass in here.